bindings/cpp/include/data_structure.hpp

```diff
@@ -39,17 +39,143 @@ enum class EntryMode : int {
 /**
  * @struct Metadata
  * @brief The metadata of a file or directory
+ *
+ * Metadata contains all the information related to a specific path.
+ * Depending on the context of the requests, the metadata for the same path may
+ * vary. For example, two versions of the same path might have different content
+ * lengths. Keep in mind that metadata is always tied to the given context and
+ * is not a global state.
  */
 class Metadata {
  public:
+  // Basic file information
   EntryMode type;
   std::uint64_t content_length;
-  std::optional<std::string> cache_control;
-  std::optional<std::string> content_disposition;
-  std::optional<std::string> content_md5;
-  std::optional<std::string> content_type;
-  std::optional<std::string> etag;
-  std::optional<std::chrono::system_clock::time_point> last_modified;
+
+  // HTTP-style headers
+  std::optional<std::string>
+      cache_control;  ///< Cache-Control header as defined by RFC 7234
+  std::optional<std::string>
+      content_disposition;  ///< Content-Disposition header as defined by RFC
+                            ///< 6266
+  std::optional<std::string>
+      content_md5;  ///< Content-MD5 hash (deprecated but widely used)
+  std::optional<std::string>
+      content_type;  ///< Content-Type header as defined by RFC 9110
+  std::optional<std::string>
+      content_encoding;  ///< Content-Encoding header as defined by RFC 7231
+  std::optional<std::string> etag;  ///< ETag header as defined by RFC 7232
+  std::optional<std::chrono::system_clock::time_point>
+      last_modified;  ///< Last-Modified timestamp
+
+  // Versioning information
+  std::optional<std::string>
+      version;                     ///< Version identifier (e.g., S3 version ID)
+  std::optional<bool> is_current;  ///< Whether this is the current version
+  bool is_deleted;  ///< Whether this entry has been marked as deleted
+
+  // Default constructor
+  Metadata() : content_length(0), is_deleted(false) {}
```

Comment:
`type` must be set during constructor.

Reply:
had used cpp11 default init style to refactor it